flink-ml-lib/src/test/java/org/apache/flink/ml/evaluation/BinaryClassificationEvaluatorTest.java

```diff
@@ -0,0 +1,245 @@
+                        .as("label", "rawPrediction");
+        inputDataTableWithWeight =
+                tEnv.fromDataStream(env.fromCollection(INPUT_DATA_WITH_WEIGHT))
+                        .as("label", "rawPrediction", "weight");
+    }
+
+    @Test
+    public void testParam() {
+        BinaryClassificationEvaluator binaryEval = new BinaryClassificationEvaluator();
+        assertEquals("label", binaryEval.getLabelCol());
+        assertNull(binaryEval.getWeightCol());
+        assertEquals("rawPrediction", binaryEval.getRawPredictionCol());
+        assertArrayEquals(
+                new String[] {"areaUnderROC", "areaUnderPR"}, binaryEval.getMetricsNames());
+        binaryEval
+                .setLabelCol("labelCol")
+                .setRawPredictionCol("raw")
+                .setMetricsNames(BinaryClassificationEvaluatorParams.AREA_UNDER_ROC)
+                .setWeightCol("weight");
+        assertEquals("labelCol", binaryEval.getLabelCol());
+        assertEquals("weight", binaryEval.getWeightCol());
+        assertEquals("raw", binaryEval.getRawPredictionCol());
+        assertArrayEquals(new String[] {"areaUnderROC"}, binaryEval.getMetricsNames());
+    }
+
+    @Test
+    public void testSaveLoadAndEvaluate() throws Exception {
+        BinaryClassificationEvaluator eval =
+                new BinaryClassificationEvaluator()
+                        .setMetricsNames("areaUnderPR", "ks", "areaUnderROC");
+        BinaryClassificationEvaluator loadedEval =
+                StageTestUtils.saveAndReload(tEnv, eval, tempFolder.newFolder().getAbsolutePath());
+        Table evalResult = loadedEval.transform(inputDataTable)[0];
+        assertArrayEquals(
+                new String[] {"areaUnderPR", "ks", "areaUnderROC"},
+                evalResult.getResolvedSchema().getColumnNames().toArray());
+        List<Row> results = IteratorUtils.toList(evalResult.execute().collect());
+        Row result = results.get(0);
+        for (int i = 0; i < EXPECTED_DATA.length; ++i) {
+            assertEquals(EXPECTED_DATA[i], result.getFieldAs(i), EPS);
+        }
+    }
+
+    @Test
+    public void testEvaluate() throws Exception {
+        BinaryClassificationEvaluator eval =
+                new BinaryClassificationEvaluator()
+                        .setMetricsNames("areaUnderPR", "ks", "areaUnderROC");
+        Table evalResult = eval.transform(inputDataTable)[0];
+        List<Row> results = IteratorUtils.toList(evalResult.execute().collect());
+        assertArrayEquals(
+                new String[] {"areaUnderPR", "ks", "areaUnderROC"},
+                evalResult.getResolvedSchema().getColumnNames().toArray());
+        Row result = results.get(0);
+        for (int i = 0; i < EXPECTED_DATA.length; ++i) {
+            assertEquals(EXPECTED_DATA[i], result.getFieldAs(i), EPS);
+        }
+    }
+
+    @Test
+    public void testEvaluateWithDoubleRaw() throws Exception {
```

Comment:
Thanks for adding this test. In Spark's BinaryClassification algorithm, it also tests when label's data type is any possible numeric type, like int, shorts or decimal. Could we also add these tests?

Reply:
I have add test for label‘s data type is int or double. I think it's not needed test all the types.